synapse/push/push_tools.py

```diff
@@ -24,7 +23,7 @@ async def get_badge_count(store: DataStore, user_id: str, group_by_room: bool) -
     invites = await store.get_invited_rooms_for_local_user(user_id)
     joins = await store.get_rooms_for_user(user_id)
 
-    my_receipts_by_room = await store.get_receipts_for_user(user_id, ReceiptTypes.READ)
+    my_receipts_by_room = await store.get_receipts_for_user(user_id)
```

Comment:
I'm not sure if the changes to the various store methods to remove the receipt makes sense? I think it makes an assumption that all receipts are some sort of read-receipt, which might not be true?

It might make more sense to have `get_receipts_for_user` and friends to take a list of receipt types or to return a map of receipt type -> latest.

Reply:
At the moment all receipts are some sort of a read receipt, so the assumption is true, at least right now, so I am a little hesitant to make changes (especially since I am not that confident with the SQL stuff). That being said, I can certainly give it a shot, if you think it's worth it

Reply:
Yeah... let me think a bit more about it, but that does seem silly to abstract it for no reason.

Reply:
I'm going to consolidate this conversation into https://github.com/matrix-org/synapse/pull/12168/files#r829202648

Reply:
:+1: I am going to mark this one as resolved